assume/markets/clearing_algorithms/contracts.py

```diff
@@ -0,0 +1,659 @@
+# SPDX-FileCopyrightText: ASSUME Developers
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+import asyncio
+import logging
+from datetime import datetime, timedelta
+from itertools import groupby
+from operator import itemgetter
+from typing import Callable
+
+import pandas as pd
+from dateutil import rrule as rr
+from dateutil.relativedelta import relativedelta as rd
+
+from assume.common.market_objects import (
+    ClearingMessage,
+    MarketConfig,
+    MarketProduct,
+    MetaDict,
+    Order,
+    Orderbook,
+)
+from assume.markets.base_market import MarketRole
+
+log = logging.getLogger(__name__)
+
+
+class PayAsBidContractRole(MarketRole):
```

Comment:
would it be also a good idea to have some description docstrings here? for the automated documentation? or the one in text is sufficient from your point fo view?

Reply:
I added docstrings as well, thanks